tests/benchmarks/bench_assembly.cpp

```diff
@@ -0,0 +1,389 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "bench.hpp"
+
+#ifdef MFEM_USE_BENCHMARK
+
+/*
+  This benchmark contains the implementation of the CEED's bake-off problems:
+  high-order kernels/benchmarks designed to test and compare the performance
+  of high-order codes.
+
+  See: ceed.exascaleproject.org/bps and github.com/CEED/benchmarks
+
+   * --benchmark_filter=[SetupBP/BP/BK][1-6][PARTIAL/ELEMENT/FULL]/[1-max_order]
+   * --benchmark_context=device=[cpu/cuda/hip]
+*/
+
+// The maximum polynomial order used for benchmarking
+const int max_order = 6;
+// The maximum number of dofs for benchmarking
+const int max_dofs = 1e7;
+
+struct BakeOff
```

Comment:
BakeOff here is misleading: CEED does not have assembly bakeoffs problems or kernels.

Reply:
We are not only benchmarking the assembly, but I'm open to any other name if you have a better proposition.